src/app/src/main/kotlin/org/vaccineimpact/api/app/repositories/jooq/JooqResponsibilitiesRepository.kt

```diff
@@ -123,6 +123,25 @@ class JooqResponsibilitiesRepository(
         }
     }
 
+    override fun getResponsibilityId(groupId: String, touchstoneVersionId: String, scenarioId: String): Int
+    {
+        val responsibilitySet = getResponsibilitySet(groupId, touchstoneVersionId)
+        if (responsibilitySet != null)
+        {
+            val id = dsl.select(Tables.RESPONSIBILITY.ID)
+                    .fromJoinPath(Tables.RESPONSIBILITY, Tables.SCENARIO, Tables.SCENARIO_DESCRIPTION)
+                    .where(Tables.RESPONSIBILITY.RESPONSIBILITY_SET.eq(responsibilitySet[Tables.RESPONSIBILITY_SET.ID])
+                    .and(Tables.SCENARIO_DESCRIPTION.ID.eq(scenarioId))
+            ).singleOrNull() ?: throw UnknownObjectError(scenarioId, "responsibility")
```

Comment:
Maybe use `fetchAnyInto` to do the `singleOrNull` and `into` all in one step? Also, the indentation is a bit screwy here.

Reply:
Does `fetchAnyInto` return null if there is no record or does it throw? Wasn't sure

Reply:
Not sure, tbh